net-p2p/go-ipfs/go-ipfs-0.4.21.ebuild

```diff
@@ -0,0 +1,63 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit bash-completion-r1 golang-vcs-snapshot systemd user
+
+DESCRIPTION="Main implementation of IPFS"
+HOMEPAGE="https://ipfs.io/"
+SRC_URI="https://dist.ipfs.io/go-ipfs/v${PV}/go-ipfs-source.tar.gz -> ${P}.tar.gz
+	https://raw.githubusercontent.com/ipfs/go-ipfs/v${PV}/misc/completion/ipfs-completion.bash -> ${P}.bash"
+EGO_PN="github.com/ipfs/go-ipfs"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="sys-fs/fuse"
+#S="${WORKDIR}/go-ipfs"
+
+S="${WORKDIR}/${P}/src/${EGO_PN}"
+
+src_compile() {
+	local mygoargs=(
+		-v
+		-work
+		-x
+		-tags release
+		-o ipfs ${EGO_PN}/cmd/ipfs
+	)
+
+	GOPATH="${WORKDIR}/${P}:$(get_golibdir_gopath)" \
+		go build "${mygoargs[@]}" || die
+}
+
+src_test() {
+	GOPATH="${WORKDIR}/${P}:$(get_golibdir_gopath)" \
+		go test -v -work -x ${EGO_PN}/cmd/ipfs/... || die
+}
+
+src_install() {
+	dobin ipfs
+
+	systemd_dounit "${FILESDIR}/ipfs.service"
+	newinitd "${FILESDIR}/ipfs.init" ipfs
+	newconfd "${FILESDIR}/ipfs.confd" ipfs
+
+	newbashcomp "${DISTDIR}/${P}.bash" "ipfs"
+	keepdir /var/log/ipfs
+}
+
+pkg_preinst() {
+	enewgroup ipfs
```

Comment:
(This is related to your other PR, just follow the instructions there, and lets update this ebuild when the proper group/user additions are done.)